lib/connections/limiter.go

```diff
@@ -97,52 +173,71 @@ func (lim *limiter) String() string {
 	return "connections.limiter"
 }
 
+func (lim *limiter) newLimitedReader(remoteID protocol.DeviceID, r io.Reader, isLAN bool) io.Reader {
+	lim.mu.Lock()
+	defer lim.mu.Unlock()
+	deviceLimiter := lim.getReadLimiter(remoteID)
+	return &limitedReader{reader: r, limiter: lim, deviceLimiter: deviceLimiter, isLAN: isLAN}
+}
+
+func (lim *limiter) newLimitedWriter(remoteID protocol.DeviceID, w io.Writer, isLAN bool) io.Writer {
+	lim.mu.Lock()
+	defer lim.mu.Unlock()
+	deviceLimiter := lim.getWriteLimiter(remoteID)
+	return &limitedWriter{writer: w, limiter: lim, deviceLimiter: deviceLimiter, isLAN: isLAN}
```

Comment:
The fact that processblabla does locking outside the function, and this does inside, feels wrong.

Reply:
We are starting to contradict each other once again x-): My reasoning to suggest this was that the locking now happens in `CommitConfiguration` and in these `newLimited...` functions, which are all "top-level", i.e. called "externally" (not all outside of the package, but outside of the `limiter` code). Also locking the entirety of `CommitConfiguration` was considered necessary to disallow concurrent change and thus with a single lock the locking cannot happen on `get...Limiter` anymore (single lock is not a requirement obviously, I just consider it cleaner).

Reply:
Fine, but why can't we lock lim.mu before calling the newblabla twice in the connection service, this way we lock the same way in both  places. Also a function that needs locking before called, we should add the locked word into the name like we do in the model.

Reply:
Now I see.  
@qepasa So locking should happen [here](https://github.com/syncthing/syncthing/pull/4603/files/6967960f5032d4fa89843731f945bfd8f4daceef#diff-d31f92d68068bfede5c6b6b01e1b558bR268) instead and a whole bunch of functions needs the `Locked` suffix: `setLimits`, `new...`, `get...`, `processD...`.